include/ppx/application.h

```diff
@@ -34,24 +35,13 @@
 #include <GLFW/glfw3.h>
 
 #if defined(PPX_ANDROID)
-    #define SETUP_APPLICATION(AppType)                  \
-        AppType app;                                    \
-        bool InitVulkan(android_app* androidContext)    \
-        {                                               \
-            app.SetAndroidContext(androidContext);      \
-            return true;                                \
-        }                                               \
-        void DeleteVulkan(void)                         \
-        {                                               \
-        }                                               \
-        bool IsVulkanReady(void)                        \
-        {                                               \
-            return app.GetAndroidContext() != nullptr;  \
-        }                                               \
-        bool VulkanDrawFrame(int argc, char** argv)     \
-        {                                               \
-            int res = app.Run(argc, argv);              \
-            return res;                                 \
+    #define SETUP_APPLICATION(AppType)                                         \
+        AppType app;                                                           \
+        bool AndroidMain(android_app* androidContext, int argc, char** argv)   \
```

Comment:
Do we need two functions that have Android and Main/main in it? Can we not find a way to combine this and the `android_main` function? To reduce the possibility of confusion.

Reply:
imo, it should be 
```
#define SETUP_APPLICATION(AppType) \
    Application* GetApp()          \
    {                              \
        static AppType app;        \
        return app;                \
    }
```
and use by both desktop and mobile build

Reply:
Agree with the sentiment...just need to think through a bit more if forcing the App to be a static varaible will have any particular implications with order of initialization.

Reply:
Global static has even worse initialization ordering issue.
Anything static in function should be initialized at first call time, I do wish the init once and never call destructor wrapper exist in bigwheels, but I'm not about to add all of that in a single PR.

Reply:
I don't really want myself or anyone debugging initializatin ordering issues when using BigWheels. All non-simple initialization needs to happen when main() is called not before.

Reply:
Well, now that there's only one function using it, it can just be a local variable.